### PR TITLE
Feat: Added cartContext

### PR DESCRIPTION
**Pull request by RahulSingh9131**
This branch contains
1. cart context for the cart items
2. some minor refactoring in productListing and Header.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -5,17 +5,20 @@ import App from "./App";
...
-        <App />
-      </Router>
-    </FilterProvider>
+    <CartProvider>
+      <FilterProvider>
+        <Router>
+          <App />
+        </Router>
+      </FilterProvider>
+    </CartProvider>
```
Comment (rahulcs754) [inline src/index.js]: you can make a Allprovider like as you router

**Review by rahulcs754 — APPROVED** [review]
Code looks good.
just small suggestion

File: src/index.js [hunk 2]
```diff
@@ -5,17 +5,20 @@ import App from "./App";
...
 ReactDOM.render(
   <React.StrictMode>
-    <FilterProvider>
-      <Router>
-        <App />
-      </Router>
-    </FilterProvider>
+    <CartProvider>
+      <FilterProvider>
+        <Router>
```
Comment (aishwarya-bm) [inline src/index.js]: The router should be enclosing all the providers and components

File: src/context/CartContext.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { createContext,useContext,useReducer } from "react";
+
+const CartContext=createContext();
+
+const CartProvider=({children})=>{
+
+    const cartReducerFunc=(cartState,action)=>{
```
Comment (aishwarya-bm) [inline src/context/CartContext.js]: reducer can be extracted out and can be imported wherever required

**Review by aishwarya-bm — APPROVED** [review]
clean and readable code, good naming conventions. Left a few suggestions.

File: src/context/CartContext.js (hunk 3 above)
Comment (RahulSingh9131) [reply src/context/CartContext.js]: sure will do this.

File: src/index.js (hunk 2 above)
Comment (RahulSingh9131) [reply src/index.js]: ok will make the change.

File: src/index.js (hunk 1 above)
Comment (RahulSingh9131) [reply src/index.js]: ok will do that.

**Pull request merged**